pkg/linear-pools/contracts/midas-linear-pool/MidasLinearPoolRebalancer.sol

```diff
@@ -18,62 +18,48 @@ pragma experimental ABIEncoderV2;
 import "./interfaces/ICToken.sol";
 
 import "@balancer-labs/v2-interfaces/contracts/pool-utils/ILastCreatedPoolFactory.sol";
+//import "@balancer-labs/v2-interfaces/contracts/solidity-utils/openzeppelin/IERC20.sol";
 
 import "@balancer-labs/v2-solidity-utils/contracts/math/FixedPoint.sol";
 import "@balancer-labs/v2-solidity-utils/contracts/openzeppelin/ERC20.sol";
 import "@balancer-labs/v2-solidity-utils/contracts/openzeppelin/SafeERC20.sol";
 
-import "./contracts/LinearPoolRebalancer.sol";
+import "@balancer-labs/v2-pool-linear/contracts/LinearPoolRebalancer.sol";
 
 contract MidasLinearPoolRebalancer is LinearPoolRebalancer {
-    using FixedPoint for uint256;
     using SafeERC20 for IERC20;
+    using FixedPoint for uint256;
 
     uint256 private immutable _divisor;
 
     // These Rebalancers can only be deployed from a factory to work around a circular dependency: the Pool must know
     // the address of the Rebalancer in order to register it, and the Rebalancer must know the address of the Pool
     // during construction.
-    constructor(IVault vault, IBalancerQueries queries)
-        LinearPoolRebalancer(ILinearPool(ILastCreatedPoolFactory(msg.sender).getLastCreatedPool()), vault, queries)
-    {
-        ILinearPool pool = ILinearPool(ILastCreatedPoolFactory(msg.sender).getLastCreatedPool());
-        ERC20 mainToken = ERC20(address(pool.getMainToken()));
-        ERC20 wrappedToken = ERC20(address(pool.getWrappedToken()));
-
-        // The CToken function exchangeRateCurrent returns the rate scaled to 18 decimals.
+    constructor(
+        IVault vault,
+        IBalancerQueries queries,
+        address mainToken,
+        address wrappedToken
```

Comment:
Instead of passing these into the constructor, I would suggest employing a similar trick to the [`GearboxLinearPoolRebalancer`](https://github.com/orbcollective/linear-pools/blob/c79ec30158e1b0072d159183c8b62740a8ef81c3/pkg/linear-pools/contracts/gearbox-linear-pool/GearboxLinearPoolRebalancer.sol#L33), where we query the tokens directly from the deployed pool.